sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -68,15 +119,128 @@ public Calc copy(RelTraitSet traitSet, RelNode input, RexProgram program) {
           BeamCalcRel.class.getSimpleName(),
           pinput);
       PCollection<Row> upstream = pinput.get(0);
+      Schema outputSchema = CalciteUtils.toSchema(getRowType());
+
+      final JavaTypeFactory typeFactory =
+          new JavaTypeFactoryImpl(BeamRelDataTypeSystem.INSTANCE) {
+            @Override
+            public Type getJavaClass(RelDataType type) {
+              if (type instanceof BasicSqlType || type instanceof IntervalSqlType) {
+                if (type.getSqlTypeName() == SqlTypeName.FLOAT) {
+                  return type.isNullable() ? Float.class : float.class;
+                }
+              }
+              return super.getJavaClass(type);
+            }
+          };
+      final BlockBuilder builder = new BlockBuilder();
+
+      final PhysType physType =
+          PhysTypeImpl.of(typeFactory, getRowType(), JavaRowFormat.ARRAY, false);
+
+      Expression input =
+          Expressions.convert_(Expressions.call(processContextParam, "element"), Row.class);
+
+      final RexBuilder rexBuilder = getCluster().getRexBuilder();
+      final RelMetadataQuery mq = RelMetadataQuery.instance();
+      final RelOptPredicateList predicates = mq.getPulledUpPredicates(getInput());
+      final RexSimplify simplify = new RexSimplify(rexBuilder, predicates, false, RexUtil.EXECUTOR);
+      final RexProgram program = BeamCalcRel.this.program.normalize(rexBuilder, simplify);
+
+      Expression condition =
+          RexToLixTranslator.translateCondition(
+              program,
+              typeFactory,
+              builder,
+              new InputGetterImpl(input, upstream.getSchema()),
+              null);
+
+      List<Expression> expressions =
+          RexToLixTranslator.translateProjects(
+              program,
+              typeFactory,
+              builder,
+              physType,
+              DataContext.ROOT,
+              new InputGetterImpl(input, upstream.getSchema()),
+              null);
+      assert expressions.size() == outputSchema.getFieldCount();
```

Comment:
should this be an exception?

Reply:
It is something that should never be false. I will just delete it.